File: 2563_count_the_number_of_fair_pairs_april.cpp
class Solution(object):
    def countFairPairs(self, a, p, q):
        a.sort()
        return self.g(a, q) - self.g(a, p - 1)
    
    def g(self, b, t):
        z, j = 0, len(b) - 1
        for i in range(len(b)):
            while i < j and b[i] + b[j] > t:
                j -= 1
            if i < j:
                z += j - i
        return z

File: 2563_count_the_number_of_fair_pairs_april.java
class Solution {
    public long countFairPairs(int[] b, int p, int q) {
        Arrays.sort(b);
        return f(b, q) - f(b, p - 1);
    }
    long f(int[] y, int t) {
        long c = 0;
        for (int i = 0, j = y.length - 1; i < j; ++i) {
            while (i < j && y[i] + y[j] > t) j--;
            c += j - i;
        }
        return c;
    }
}
